pkg/kubelet/network/dns/dns.go

```diff
@@ -241,19 +247,96 @@ func (c *Configurer) GetPodDNS(pod *v1.Pod) (*runtimeapi.DNSConfig, error) {
 			return nil, err
 		}
 	}
-	useClusterFirstPolicy := ((pod.Spec.DNSPolicy == v1.DNSClusterFirst && !kubecontainer.IsHostNetworkPod(pod)) || pod.Spec.DNSPolicy == v1.DNSClusterFirstWithHostNet)
-	if useClusterFirstPolicy && len(c.clusterDNS) == 0 {
+	return &runtimeapi.DNSConfig{
+		Servers:  hostDNS,
+		Searches: hostSearch,
+		Options:  hostOptions,
+	}, nil
+}
+
+func (c *Configurer) getPodDNSType(pod *v1.Pod) podDNSType {
+	dnsPolicy := pod.Spec.DNSPolicy
+	if dnsPolicy == v1.DNSNone {
+		if utilfeature.DefaultFeatureGate.Enabled(features.CustomPodDNS) {
+			return podDNSNone
+		}
+		// Fallback to DNSDefault. This should not happen as kube-apiserver should
+		// have rejected setting dnsPolicy to DNSNone when feature gate is disabled.
+		return podDNSHost
```

Comment:
fall back to cluster-first?

Reply:
Sure, I forgot `DNSDefault` is not the default...